ocdskingfisherarchive/collection.py

```diff
@@ -94,7 +131,32 @@ def write_data_file(self):
         if return2 != 0:
             raise Exception(command2 + ' Got Return ' + str(return2))
 
+        os.unlink(filename)
+
         return filename+'.lz4'
 
     def get_s3_directory(self):
         return self.source_id + '/' + str(self.data_version.year) + '/' + str(self.data_version.month).zfill(2)
+
+    def delete_data_files(self):
+        if self.get_data_files_exist():
+            data_dir = self.config.directory_data + '/' + self.source_id + '/' + \
+                self.data_version.strftime("%Y%m%d_%H%M%S")
+            # We use os.system here so we know the exact command so we can set up sudo correctly
+            return1 = os.system('sudo -u ocdskfs /bin/rm -rf '+data_dir)
```

Comment:
Why not just set things up so that the ocdskfs user runs the archive command? Seems like a lot of trouble otherwise (changes to the sudo command require changes to deploy...).

Reply:
So this is actually matching the current setup as closely as possible, to minimise changes on server. We already have sudo set up: https://github.com/open-contracting/deploy/blob/master/salt/ocdskingfisherarchive/archive.sudoers

And also noting  that we have to have some salt changes when deploying this (config file, logging config file, new pillar variables, pip); I've already done that initial work and can present it soon. But minimising changes for now seemed sensible.